autoblocks/_impl/tracer.py

```diff
@@ -101,38 +110,119 @@ def start_span(self):
                 props["span_id"] = prev_span_id
             self.set_properties(props)
 
+    def _evaluation_to_json(evaluation: Evaluation, evaluator_external_id: str) -> Dict[str, Any]:
+        return dict(
+            id=str(uuid.uuid4()),
+            score=evaluation.score,
+            threshold=dataclasses.asdict(evaluation.threshold) if evaluation.threshold else None,
+            metadata=evaluation.metadata,
+            evaluatorExternalId=evaluator_external_id,
+        )
+
+    async def _evaluate_event(
+        self, event: TracerEvent, evaluator: BaseEventEvaluator
+    ) -> Optional[Union[Evaluation, None]]:
+        """
+        Evaluates an event using a provided evaluator.
+        """
+        evaluation = None
+        if inspect.iscoroutinefunction(evaluator.evaluate_event):
+            try:
+                evaluation = await evaluator.evaluate_event(event=event)
+            except Exception as err:
+                log.error("Unable to execute evaluation. Error: %s", repr(err), exc_info=True)
+        else:
+            try:
+                ctx = contextvars.copy_context()
+                evaluation = await global_state.event_loop().run_in_executor(
+                    None,
+                    ctx.run,
+                    evaluator.evaluate_event,
+                    event,
+                )
+            except Exception as err:
+                log.error("Unable to execute evaluation. Error: %s", repr(err), exc_info=True)
+
+        return evaluation
+
+    async def _run_evaluators(
+        self, evaluators: List[BaseEventEvaluator], event: TracerEvent, max_evaluator_concurrency: int
+    ) -> List[Dict[str, Any]]:
+        if len(evaluators) == 0:
+            return []
+        try:
+            evaluations: List[Union[Evaluation, None]] = await gather_with_max_concurrency(
+                max_evaluator_concurrency,
+                [
+                    self._evaluate_event(
+                        event=event,
+                        evaluator=evaluator,
+                    )
+                    for evaluator in evaluators
+                ],
+            )
+            return [
+                self._evaluation_to_json(event_evaluation=item, evaluator_external_id=evaluator.id)
+                for evaluator, item in zip(evaluators, evaluations)
+                if item is not None
+            ]
+
+        except Exception as err:
+            log.error("Unable to complete evaluating events. Error: %s", str(err), exc_info=True)
```

Comment:
up above you're using `repr` instead of `str`, can we be consistent? also do we need either?